### PR TITLE
Contributing guidelines

### PR DESCRIPTION
**Pull request by andyjsbell**
# Summary
- **RFCs**: [Link to RFC](#./link/to/rfc), [Link to RFC](#./link/to/rfc), or $\emptyset$.
- **Categories**: any of `protocol-units`, `networks`, `scripts`, `util`, `cicd`, or `misc`.

Two documents to be added: Contributing and Code of Conduct

# Changelog

# Testing

Read for accuracy, tested links and typos.
# Outstanding issues

### CODE REVIEW

**Review by 0xPrimata — APPROVED** [review]

**Review by 0xmovses — APPROVED** [review]
Checked for typos, looks good.
👍

**Review by andygolay — APPROVED** [review]
:shipit:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+- **Module Organization**: Keep modules organized logically, typically grouping related structs, enums, and functions together.
+- **Error Handling**: Prefer `Result` and `Option` types over panics where possible.
+- **Documentation Comments**: Use `///` for public-facing functions and modules, and `//` for inline comments.
+- **Linting and Formatting**: Ensure your code passes `cargo fmt` and `cargo clippy` checks.
+
+#### Documentation Style
+
+When documenting Rust code, follow these conventions:
+
+- **Doc Comments**: Use `///` for public API documentation, describing the purpose, parameters, and return values.
```
Comment (apenzk) [inline CONTRIBUTING.md:184]: this could do with an example to e.g. standardise the function comments

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+For Rust, we follow the standard Rust style guidelines, which include the following:
+
+- **Indentation**: Use 4 spaces for indentation, no tabs.
+- **Naming Conventions**: 
+  - Use `snake_case` for variables, functions, and file names.
+  - Use `PascalCase` for structs, enums, and traits.
+  - Use `ALL_CAPS` with underscores for constants.
+- **Module Organization**: Keep modules organized logically, typically grouping related structs, enums, and functions together.
+- **Error Handling**: Prefer `Result` and `Option` types over panics where possible.
+- **Documentation Comments**: Use `///` for public-facing functions and modules, and `//` for inline comments.
```
Comment (apenzk) [inline CONTRIBUTING.md:177]: this could do with an example to e.g. standardise the function comments

**Review by franck44 — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (l-monninger) [reply CONTRIBUTING.md:177]: These are already standardized in Rust though.

Comment (andyjsbell) [reply CONTRIBUTING.md:177]: Fair point, makes sense to link to https://doc.rust-lang.org/style-guide/index.html

**Review by radupopa369 — APPROVED** [review]

**Review by mzabaluev — APPROVED** [review]

**Pull request merged**